draft-ietf-gnap-core-protocol.md

```diff
@@ -771,72 +772,66 @@ The request MUST be sent as a JSON object in the body of the HTTP
+field. This field MUST be an object (for a [single access token](#request-token-single)) or
+an array of these objects (for [multiple access tokens](#request-token-multiple)), 
+as described in the following sections.
 
-### Requesting a Single Access Token {#request-resource-single}
+### Requesting a Single Access Token {#request-token-single}
 
-When requesting an access token, the client instance MUST send a
-`resources` field containing a JSON array. The elements of the JSON
-array represent rights of access that the client instance is requesting in
-the access token. The requested access is the union of all elements
-within the array.
+To request a single access token, the client instance sends an `acccess_token` object
+composed of the following fields.
 
-The client instance declares what access it wants to associate with the
-resulting access token using objects that describe multiple
-dimensions of access. Each object contains a `type`
-property that determines the type of API that the client instance is calling.
+access (array of objects/strings)
+: Describes the rights that the client instance is requesting for one or more access tokens to be
+    used at RS's. This field is REQUIRED. {{resource-access-rights}}
+
+label (string)
+: A unique name chosen by the client instance to refer to the resulting access token. If this field
+    is included in the request, the AS MUST include the same label in the [token response](#response-token).
+    This field is REQUIRED if used as part of a [multiple access token request](#request-token-multiple),
+    and is OPTIONAL otherwise.
+
+flags (array of strings)
+: A set of flags that indicate desired attributes or behavior to be attached to the access token by the
+    AS. This field is OPTIONAL.
+
+The values of the `flags` field defined by this specification are as follows:
+
+bearer
+: If this flag is included, the access token being requested is a bearer token.
+    If this flag is omitted, the access token is bound to the key used
+    by the client instance in this request, or the key's most recent rotation. 
+    Methods for presenting bound and bearer access tokens are described
+    in {{use-access-token}}.
+    \[\[ [See issue #38](https://github.com/ietf-wg-gnap/gnap-core-protocol/issues/38) \]\]
 
-type (string)
-: The type of resource request as a string. This field MAY
-      define which other fields are allowed in the request object. 
-      This field is REQUIRED.
+split
+: If this flag is included, the client instance is capable of 
+    receiving a different number of tokens than specified in the
+    [token request](#request-token), including 
+    receiving [multiple access tokens](#response-token-multiple)
+    in response to any [single token request](#request-token-single) or a
+    different number of access tokens than requested in a [multiple access token request](#request-token-multiple).
+    The `label` fields of the returned additional tokens are chosen by the AS. 
```

Comment:
We should say explicitly that the `label` is opaque, and must not be used by the AS to denote the usage of access tokens (in the "split" case), otherwise they will do just that.